src/main/java/soot/asm/AsmMethodSource.java

```diff
@@ -2373,11 +2373,17 @@ protected void tryCorrectingLocalNames(final Jimple jimp, final JimpleBody jb) {
               for (AbstractInsnNode i = lvn.start.getPrevious(); (uStart = units.get(i)) == null;) {
                 i = i.getNext();
               }
+              if (uStart instanceof UnitContainer) {
+                uStart = ((UnitContainer) uStart).getFirstUnit();
+              }
               // Get the previous real instruction before 'end'
               Unit uEnd;
               for (AbstractInsnNode i = lvn.end.getPrevious(); (uEnd = units.get(i)) == null;) {
                 i = i.getPrevious();
               }
+              if (uEnd instanceof UnitContainer) {
```

Comment:
As above, it makes sense that there could be a `UnitContainer` here. However, in this case, `uEnd` is supposed to be assigned with the final Unit in the range defined by the `LocalVariableNode` so it seems unlikely to me that using `getFirstUnit()` is appropriate here. I would assume getting the last Unit in the `UnitContainer` would be the correct approach for this case. However, without a test case for this PR that utilizes this new code, I can't say for sure.

Reply:
The merge only happens here https://github.com/soot-oss/soot/blob/478ba74647dad3be72b696c999a7debeddedc754/src/main/java/soot/asm/AsmMethodSource.java#L1850 and it ensures there is only one local when converting instructions on different paths but actually the same local. The firstUnit is then used to decide the local: https://github.com/soot-oss/soot/blob/a2eb26fded4a4a7107aa3530635ab6a3ca0127ad/src/main/java/soot/asm/StackFrame.java#L174

So I think this should be correct.